File: src/main/java/br/campos/poi/model/abst/TypedData.java
package br.campos.poi.model.abst;

import br.campos.poi.util.DataType;

/**
 * Abstração de Objetos que podem ser convertidos pela AbstractDataFactory
 * 
 * @author Caio
 */
public abstract class TypedData {

	/**
	 * Define o tipo de dado o TypedData possui
	 * 
	 * @return Enum com o tipo de dado
	 */
	public DataType getType() {
		return DataType.NO_TYPE;
	}
}

File: src/main/kotlin/br/campos/poi/model/abst/TypedData.kt
package br.campos.poi.model.abst

import br.campos.poi.util.DataType

/**
 * Abstração de Objetos que podem ser convertidos pela AbstractDataFactory
 *
 * @author Caio
 */
abstract class TypedData {

	/**
	 * Define o tipo de dado que o TypedData possui
	 */
	open val type: DataType
		get() = DataType.NO_TYPE
}
